test/server/http/admin_test.cc

```diff
@@ -49,6 +127,146 @@ class AdminFilterTest : public testing::TestWithParam<Network::Address::IpVersio
   Http::TestHeaderMapImpl request_headers_;
 };
 
+INSTANTIATE_TEST_CASE_P(IpVersions, AdminStatsTest,
+                        testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                        TestUtility::ipTestParamsToString);
+
+TEST_P(AdminStatsTest, StatsAsJson) {
+  InSequence s;
+  store_->initializeThreading(main_thread_dispatcher_, tls_);
+
+  Stats::Histogram& h1 = store_->histogram("h1");
+  Stats::Histogram& h2 = store_->histogram("h2");
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 200));
+  h1.recordValue(200);
+
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h2), 100));
+  h2.recordValue(100);
+
+  store_->mergeHistograms([]() -> void {});
+
+  // Again record a new value in h1 so that it has both interval and cumulative values.
+  // h2 should only have cumulative values.
+  EXPECT_CALL(sink_, onHistogramComplete(Ref(h1), 100));
+  h1.recordValue(100);
+
+  store_->mergeHistograms([]() -> void {});
+
+  EXPECT_CALL(*this, free(_));
+
+  std::map<std::string, uint64_t> all_stats;
+
+  std::string actual_json = statsAsJsonHandler(all_stats, store_->histograms());
+
+  const std::string expected_json = R"EOF({
```

Comment:
This reminds me; do we always generate pretty-printed JSON?  I think it would be better to deliver it compact in production, though it's nice to have the test be pretty-printed for test.cc legibility.

Reply:
Yes. Currently we do generate pretty-printed JSON. Now I have added a parameter to control that and defaulted it to false and changed the test to print in pretty form. PTAL.